machine/machine_core/machine_virtual.py

```diff
@@ -139,15 +144,15 @@ def stdchannel_redirected(stdchannel, dest_filename):
 
 
 class VirtNetwork:
-    def __init__(self, network=None, image="generic"):
-        self.locked = []
+    def __init__(self, id: Optional[int] = None, image: str = "generic"):
+        self.locked: List[int] = []
```

Comment:
Isn't it more usual to write it like this?

```python
class VirtNetwork:
    locked: List[int]
    image: str

    def __init__(self, ...
```

Reply:
I feel like annotations for `network`, `offset`, `last`, and `hostnet` might also be helpful here.